buildSrc/src/main/java/com/google/firebase/gradle/plugins/semver/ClassInfo.kt

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins.semver
+
+import org.objectweb.asm.tree.ClassNode
+import org.objectweb.asm.tree.FieldNode
+import org.objectweb.asm.tree.MethodNode
+
+class ClassInfo(val _node: ClassNode, val _classNodes: Map<String, ClassNode>) {
+  val name: String
+  val node: ClassNode
+  val fields: Map<String, FieldNode>
+  val methods: Map<String, MethodNode>
+  val methodsCache: MutableMap<String, Map<String, MethodNode>>
+
+  init {
+    this.name = _node.name
+    this.node = _node
+    this.fields = getAllFields(_node)
+    this.methodsCache = mutableMapOf()
+    this.methods = getAllMethods(_node, _classNodes)
+  }
+
+  fun getAllFields(node: ClassNode): Map<String, FieldNode> {
+    var fields: MutableMap<String, FieldNode> = mutableMapOf()
+    for (field in node.fields) {
+      val descriptor = AccessDescriptor(field.access)
+      if (descriptor.isSynthetic() || UtilityClass.isObfuscatedSymbol(field.name)) continue
+      fields.put("${field.name}-${field.desc}", field)
+    }
+    return fields
```

Comment:
Avoid using mutable collections when possible.

This could translate to 

```kotlin
fun getAllfields(node: ClassNode) = node.fields
  .map(it.field)
  .filterNot{ AccessDescriptor(it.access).isSynthetic() || UtilityClass.isObfuscatedSymbol(it.name) }
  .associate { field -> "${field.name}-${field.desc}" to field }
```

here and below